java/examples/webhook-server/src/main/java/com/truelayer/signing/examples/webhook_server/WebhookVerificationHandler.java

```diff
@@ -0,0 +1,60 @@
+package com.truelayer.signing.examples.webhook_server;
+
+import com.truelayer.signing.Verifier;
+import io.javalin.http.BadRequestResponse;
+import io.javalin.http.Context;
+import io.javalin.http.Handler;
+import io.javalin.http.InternalServerErrorResponse;
+import okhttp3.OkHttpClient;
+import okhttp3.Request;
+import okhttp3.Response;
+import okhttp3.ResponseBody;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.Map;
+
+public class WebhookVerificationHandler implements Handler {
+
+    private final OkHttpClient httpClient;
+
+    public WebhookVerificationHandler(OkHttpClient httpClient) {
+        this.httpClient = httpClient;
+    }
+
+    @Override
+    public void handle(@NotNull Context ctx) throws Exception {
+        // extract jku from signature
+        String tlSignature = ctx.header("Tl-Signature");
+        if (tlSignature == null) throw new BadRequestResponse("Missing Tl-Signature header");
+
+        String jku = Verifier.extractJku(tlSignature);
```

Comment:
Nothing related to this PR in particular, but is there any specific reason in providing the `extractJku(...)` operation and the next  `verifyWithJwks(...)` into 2 separate methods? . Could we hide this complexity for our customers by simply giving a verify(...) method ? 

I can see this as a way to make the JKU endpoint verification optional, but I don't have enough context here

Reply:
The idea there is to keep the HTTP fetching logic out of scope for this library. By cutting down the scope we make this lib easier to maintain.

I think the HTTP fetch would be in scope for the SDK libs though, I see them offering a webhook verification method that uses _truelayer-signing_ but also takes care of fetching and caching the jwks.